tests/unit/mixins/createStatefulChildrenMixin.ts

```diff
@@ -349,5 +356,102 @@ registerSuite({
 		}).then(() => {
 			assert.isTrue(List([ widget3 ]).equals(parent.children));
 		});
+	},
+	'createChild()': {
+		'creation during mixin'() {
+			let p: Promise<[string, Renderable]>;
+			const registry = Object.create(widgetRegistry);
+			const createFoo = compose({})
+				.mixin({
+					mixin: createStatefulChildrenMixin,
+					initialize(instance) {
+						p = instance.createChild(createRenderable, <RenderableOptions> {
+							render() {
+								return h('div');
+							}
+						});
+					}
+				});
+			const foo = createFoo({
+				registryProvider: {
+					get(type: string) {
+						return type === 'widgets' ? registry : null;
+					}
+				}
+			});
+			assert(foo);
+			assert(p);
+			return p.then((result) => {
+				const [ id, widget ] = result;
+				assert.include(id, 'undefined-child');
```

Comment:
This is a side effect of stubbing things... because we aren't using a parent that has an ID, it gets back `undefined` for the parent ID, but then the `id` generated here is unique in the sense of `undefined-child-1` and any other children created from this parent will be unique.  The mixin which provides widgets with an ID when none is supplied will generate them uniquely.  But I can see where it looks a bit confusing, so I will update the test to stub a widget with an ID.